samples/rust/rust_inplace.rs

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Rust minimal in-place sample.
+
+use kernel::prelude::*;
+
+module! {
+    type: RustInPlace,
+    name: "rust_inplace",
+    author: "Rust for Linux Contributors",
+    description: "Rust minimal in-place sample",
+    license: "GPL",
+}
+
+#[pin_data(PinnedDrop)]
+struct RustInPlace {
+    numbers: Vec<i32>,
+}
```

Comment:
I think it would be beneficial to also put something in here that people can't put into non-in-place modules, e.g. a `Mutex`, of coruse `sync` would have to land first.

Reply:
I believe the main motivation for this PR is indeed to be able to use a module-wide `Mutex` without having to use ctor tricks?